drivers/platform/stm32/stm32_pwm.h

```diff
@@ -0,0 +1,109 @@
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef PWM_EXTRA_H_
+#define PWM_EXTRA_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "gpio.h"
+#include "stm32_gpio.h"
+#include "stm32_hal.h"
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdbool.h>
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/**
+ * @struct stm32_pwm_init_param
+ * @brief Structure holding the STM32 PWM parameters.
+ */
+typedef struct stm32_pwm_init_param {
+	/** PWM Timer base */
+	TIM_TypeDef *base;
+	/** PWM Timer Channel */
+	uint32_t timer_chn;
+	/** PWM Timer GPIO init parameters */
+	//stm32_gpio_init_param *gpio;
+	gpio_init_param *gpio;
+	/** Get timer clock frequency function */
+	uint32_t (*get_timer_clock)(void);
+} stm32_pwm_init_param;
+
+/**
+ * @struct stm32_pwm_desc
+ * @brief Structure holding the STM32 PWM descriptor.
+ */
+typedef struct stm32_pwm_desc {
+	/** PWM Timer Instance */
+	TIM_HandleTypeDef htimer;
+	/** Timer GPIO instance */
+	gpio_desc *gpio;
+	/** PWM Timer period */
+	uint32_t timer_period;
+	/** PWM Timer Channel */
+	uint32_t timer_chn;
+} stm32_pwm_desc;
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+
+/* Initialize the PWM generator device */
+int32_t stm32_pwm_init(struct pwm_desc **desc,
```

Comment:
I think that you already have declarations of this functions in
https://github.com/analogdevicesinc/no-OS/blob/master/include/pwm.h
please implement those.

Reply:
I have only implemented the functions which are necessary/required for STM32 PWM init. Do you expect to implement all those declared in pwm.h?

Reply:
Implementing only the functions that you need is OK, the driver can be extended if it is necessary.